#CMakeLists.txt#

```diff
@@ -0,0 +1,271 @@
+set(CMAKE_VERBOSE_MAKEFILE ON)
+
+# CMake version check
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)  
+
+# Try to find CUDA
+find_package(CUDAToolkit)
+# Check if CUDA was found
+if(CUDAToolkit_FOUND AND NOT(USE_CPU))
+    message(STATUS "CUDA found. Adding CUDA support.")
+    SET(MaCh3_GPU_ENABLED TRUE)
+    SET(CPU_ONLY FALSE)
+else()
+    message(STATUS "CUDA not found. Proceeding without CUDA support.")
+    SET(MaCh3_GPU_ENABLED FALSE)
+    SET(CPU_ONLY TRUE)
+endif()
+
+SET(MaCh3_VERSION 0.0.2)
+if(CPU_ONLY)
+  project(MaCh3 VERSION ${MaCh3_VERSION} LANGUAGES CXX)
+else()
+  project(MaCh3 VERSION ${MaCh3_VERSION} LANGUAGES CXX CUDA)
+endif()
```

Comment:
what is this file?

Reply:
Lol this is emacs making a backup, I'll delete